Using Cross-Origin-Opener-Policy HTTP header disables the back/forward cache

cdumez opened the pull request:
#### f2479794724e0b99b9bea5b78631f518bd87be1a
<pre>
Using Cross-Origin-Opener-Policy HTTP header disables the back/forward cache
<a href="https://bugs.webkit.org/show_bug.cgi?id=274310">https://bugs.webkit.org/show_bug.cgi?id=274310</a>
<a href="https://rdar.apple.com/128678196">rdar://128678196</a>

Reviewed by Per Arne Vollan.

When process-swapping on navigation response, due to the Cross-Origin-Opener-Policy HTTP header,
we would fail to create a SuspendedPageProxy for the page in the previous WebProcess, which
would essentially disable the back/forward cache.

This is not efficient and also causes issues when navigating back to a history item that was
created by JavaScript via history.pushState().

* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::commitProvisionalPage):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/ProcessSwapOnNavigation.mm:
(-[PSONNavigationDelegate _webView:navigation:didSameDocumentNavigation:]):
((ProcessSwap, NavigateBackAfterNavigatingAwayFromCOOPUsingBFCache)):

Canonical link: <a href="https://commits.webkit.org/282105@main">https://commits.webkit.org/282105@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/bb13700a6b0b7df14d714a0201ecfeaf4655b2fd

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/62104 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/41458 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/14696 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/66084 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/12649 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/64223 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/49144 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/12989 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/50019 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/8742 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/65173 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/38482 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/53786 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/30848 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/35129 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/11024 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/11580 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/56918 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/11328 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/67812 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/6047 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/11091 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/57397 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/6073 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/53737 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/57646 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/4960 "Passed tests") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/9349 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/37258 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/38342 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/39438 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/38087 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

pvollan commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/ProcessSwapOnNavigation.mm:
```diff
@@ -7744,6 +7750,85 @@ HTTPServer server({
     done = false;
 }
 
+TEST(ProcessSwap, NavigateBackAfterNavigatingAwayFromCOOPUsingBFCache)
```
Nit: `COOP` -> `CrossOriginOpenerPolicy` and `BF` -> `BackForward`?

pvollan reviewed (approved):
R=me.

cdumez committed:
Using Cross-Origin-Opener-Policy HTTP header disables the back/forward cache
https://bugs.webkit.org/show_bug.cgi?id=274310
rdar://128678196

Reviewed by Per Arne Vollan.

When process-swapping on navigation response, due to the Cross-Origin-Opener-Policy HTTP header,
we would fail to create a SuspendedPageProxy for the page in the previous WebProcess, which
would essentially disable the back/forward cache.

This is not efficient and also causes issues when navigating back to a history item that was
created by JavaScript via history.pushState().

* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::commitProvisionalPage):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/ProcessSwapOnNavigation.mm:
(-[PSONNavigationDelegate _webView:navigation:didSameDocumentNavigation:]):
((ProcessSwap, NavigateBackAfterNavigatingAwayFromCOOPUsingBFCache)):

Canonical link: https://commits.webkit.org/282105@main
